cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -1449,30 +1515,50 @@ func (p *BicepProvider) deployModule(
 	return target.Deploy(ctx, armTemplate, armParameters, tags)
 }
 
-// Gets the path to the project parameters file path
-func (p *BicepProvider) parametersTemplateFilePath() string {
-	infraPath := p.options.Path
-	parametersFilename := fmt.Sprintf("%s.parameters.json", p.options.Module)
-	return filepath.Join(p.projectPath, infraPath, parametersFilename)
-}
+type bicepModuleType string
+
+const (
+	bicepparam bicepModuleType = "bicepparam"
+	rawBicep   bicepModuleType = "rawBicep"
+)
 
 // Gets the folder path to the specified module
-func (p *BicepProvider) modulePath() string {
+func (p *BicepProvider) bicepModule() *bicepModule {
 	infraPath := p.options.Path
-	moduleFilename := fmt.Sprintf("%s.bicep", p.options.Module)
-	return filepath.Join(p.projectPath, infraPath, moduleFilename)
+	moduleName := p.options.Module
+
+	// Check if there's a <moduleName>.bicepparam first. It will be preferred over a <moduleName>.bicep
+	moduleFilename := fmt.Sprintf("%s.bicepparam", moduleName)
+	if _, err := os.Stat(moduleFilename); err == nil {
```

Comment:
I think this Stat call is wrong now because it does not contain the full path (as would be returned by `filePath` in your structure).

We should do the filepath join earlier and pass the result os.Stat.